sootup.core/src/main/java/sootup/core/typehierarchy/MethodDispatchResolver.java

```diff
@@ -181,23 +196,12 @@ public static boolean canDispatch(
   public static Optional<MethodSignature> resolveConcreteDispatch(
       View<? extends SootClass<?>> view, MethodSignature m) {
     TypeHierarchy hierarchy = view.getTypeHierarchy();
-    ClassType superClassType = m.getDeclClassType();
-    SootClass<?> startClass = view.getClass(superClassType).orElse(null);
-    ArrayList<SootClass<?>> classesInHierachyOrder = new ArrayList<>();
-
-    // search concrete method in the class itself and its super classes
-    do {
-      ClassType finalSuperClassType = superClassType;
-      SootClass<?> superClass =
-          view.getClass(superClassType)
-              .orElseThrow(
-                  () ->
-                      new ResolveException(
-                          "Did not find class " + finalSuperClassType + " in View"));
+    ClassType current = m.getDeclClassType();
+    SootClass<?> startClass = view.getClass(current).orElse(null);
+    List<SootClass<?>> classesInHierarchyOrder = findSuperClassesInclusive(view, current);
 
-      classesInHierachyOrder.add(superClass);
-
-      SootMethod concreteMethod = findConcreteMethodInSootClass(superClass, m).orElse(null);
+    for (SootClass<?> currentClass : classesInHierarchyOrder) {
+      SootMethod concreteMethod = currentClass.getMethod(m.getSubSignature()).orElse(null);
       if (concreteMethod != null && !concreteMethod.isAbstract()) {
```

Comment:
can you unify this if-else with the following one?

Reply:
will do